### PR TITLE
tests: drivers: spi loopback on more stm32 boards

### PR DESCRIPTION
**Pull request by FRASTM**
Configure the following stm32  platforms through an overlay file, for running the tests/drivers/spi/spi_loopback 

- nucleo_l152re
- stm32f3_disco

Limit the spi speed freq to run the test correctly. The SPI1 is selected: MOSI/MISO output pins are available on the platform. 

For the nucleo_l152 DMA1 channel 2 & 3 are selected for SPI1.
Connect pin D11 & D12 of CN5  (add the corresponding harness_config "fixture") to pass the test.

For the stm32f3_disco
Connect pin PA6 to PA7 on the P connector (add the corresponding harness_config "fixture") to pass the test.
For this test the config is not choosing the DMA transfer.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Review by erwango — DISMISSED** [review]

File: tests/drivers/spi/spi_loopback/testcase.yaml [hunk 1]
```diff
@@ -24,4 +24,4 @@ tests:
     extra_args: OVERLAY_CONFIG="overlay-stm32-spi-dma.conf"
     filter: CONFIG_SOC_FAMILY_STM32
     platform_allow: nucleo_g474re nucleo_f207zg nucleo_f429zi nucleo_f746zg nucleo_wb55rg
-        nucleo_wl55jc nucleo_h743zi
+        nucleo_l152re nucleo_wl55jc nucleo_h743zi
```
Comment (ABOSTM) [inline tests/drivers/spi/spi_loopback/testcase.yaml:27]: don't you need to add stm32f3_disco to platform_allow ?

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: tests/drivers/spi/spi_loopback/testcase.yaml (hunk 1 above)
Comment (FRASTM) [reply tests/drivers/spi/spi_loopback/testcase.yaml:27]: The choice here is to run this test on stm32f3_disco without DMA

**Review by gautierg-st — DISMISSED** [review]

**Review by ABOSTM — DISMISSED** [review]

**Comment by fabiobaltieri** [comment]
> Conflicting files
> tests/drivers/spi/spi_loopback/testcase.yaml 

Can you rebase? Huge yaml patch just got merged

**Comment by carlescufi** [comment]
@FRASTM please rebase

**Review by erwango — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**